src/confluent_kafka/src/Producer.c

```diff
@@ -393,6 +393,83 @@ static PyObject *Producer_flush (Handle *self, PyObject *args,
         return cfl_PyInt_FromInt(qlen);
 }
 
+static PyObject *Producer_topic_new (Handle *self, PyObject *args,
+                                 PyObject *kwargs) {
+        rd_kafka_topic_conf_t *topic_conf = NULL;
+        rd_kafka_topic_t *topic_obj = NULL;
+        const char *topic = NULL;
+        PyObject *conf = NULL;
+        Py_ssize_t pos = 0;
+        PyObject *ko = NULL, *vo = NULL;
+
+        static char *kws[] = { "topic", "conf", NULL };
+
+        if (!PyArg_ParseTupleAndKeywords(args, kwargs, "s|O", kws, &topic, &conf))
+            return NULL;
+
+	    if (!PyDict_Check(conf))
```

Comment:
Shouldn't we error here?
```suggestion
	    if (!PyDict_Check(conf)) {
	        PyErr_SetString(PyExc_TypeError, "conf must be a dictionary or options");
```

Reply:
Yes, totally right. Added it, thanks